sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -103,8 +116,11 @@ private final class GpuSemaphore(tasksPerGpu: Int) extends Logging {
   // Map to track which tasks have acquired the semaphore.
   private val activeTasks = new ConcurrentHashMap[Long, MutableInt]
 
-  def acquireIfNecessary(context: TaskContext): Unit = {
-    val nvtxRange = new NvtxRange("Acquire GPU", NvtxColor.RED)
+  def acquireIfNecessary(context: TaskContext, waitMetric: Option[GpuMetric] = None): Unit = {
+    val nvtxRange = waitMetric match {
+      case Some(m) => new NvtxWithMetrics("Acquire GPU", NvtxColor.RED, m)
```

Comment:
We have done this in a few places. Would it be better to just have `NvtxWithMetrics` have a constructor that takes a `Option[GpuMetric]` and hide it internally?

Reply:
> On a side not I noticed that data bricks photon has a "cumulative time" metric that fills the same role we have been using for "total time" but feels a bit more accurate in the naming. Just curious if it is worth refactoring/renaming to match?

I like the cumulative name better. I'll create a separate PR to propose renaming this metric.